templates/squid.conf.header.erb

```diff
@@ -5,7 +5,7 @@
 cache_mem                     <%= @cache_mem %>
 
 <% if @memory_cache_shared -%>
```

Comment:
Maybe this would need to be...
```
<% unless @memory_cache_shared.nil? -%>
```
??
Probably need a new spec test if you're going to include this change in this PR.  Alternatively, it can be left as just a 'yes'/'no' Enum for now and someone could tackle accepting booleans in a separate PR.